psl/psl-core/src/validate/validation_pipeline/validations/relation_indexes.rs

```diff
@@ -0,0 +1,86 @@
+use diagnostics::DatamodelWarning;
+use parser_database::{
+    ast::ModelId,
+    walkers::{RelationFieldWalker, Walker},
+};
+
+use crate::{datamodel_connector::RelationMode, validate::validation_pipeline::context::Context};
+
+// { x_1, x_2, ..., x_n } is left-wise included in { y_1, y_2, ..., y_m } if and only if
+// n <= m and x_i = y_i for all i in [1, n].
+fn is_left_wise_included<T>(item: &Vec<T>, group: &Vec<T>) -> bool
+where
+    T: PartialEq,
+{
+    group.iter().take(item.len()).eq(item.iter())
+}
+
+pub(super) fn validate_missing_relation_indexes(
+    _model: Walker<'_, ModelId>,
+    relation_field: RelationFieldWalker<'_>,
+    ctx: &mut Context<'_>,
+) {
+    dbg!("Inspecting: {:?}", relation_field.name());
+
+    let is_provider_mongodb = ctx
+        .datasource
+        .map(|datasource| datasource.active_provider == "mongodb")
+        .unwrap_or(false);
+
+    if is_provider_mongodb || ctx.relation_mode != RelationMode::Prisma {
+        return;
+    }
+
+    if let Some(fields) = relation_field.referenced_fields() {
+        // vector of ids of the fields that should be part of an index in the given model, w.r.t. to left-wise inclusion.
+        let relation_fields = fields.map(|field| field.field_id()).collect::<Vec<_>>();
+
+        // for index_walker in model.indexes() {
+        //     dbg!("exploring index: {:?}", index_walker.name());
+        //     index_walker.fields().for_each(|index| {
+        //         dbg!("  index: {:?}", index);
+        //     });
+        // }
+
+        // vector of all @unique/@@unique attributes in the given model
+        // TODO: figure out how to retrieve these attributes
+        let unique_sets: Vec<Vec<_>> = vec![];
+
+        // vector of all @@index attributes in the given model
+        // TODO: figure out how to retrieve these attributes
+        let index_sets: Vec<Vec<_>> = vec![];
```

Comment:
I need to figure out this part, apparently the `unique_criterias` method is not enough